src/codecs/openexr.rs

```diff
@@ -0,0 +1,489 @@
+use crate::buffer_::{Rgb32FImage, Rgba32FImage};
+use std::convert::TryInto;
+
+
+// TODO this could be a generic function that works for any format
+/// Read the file from the specified path into an `Rgba32FImage`.
+// TODO progress? load rect?
+pub fn read_as_rgba_image_from_file(path: impl AsRef<Path>) -> ImageResult<Rgba32FImage> {
+    read_as_rgba_image(BufReader::new(std::fs::File::open(path)?))
+}
+
+// TODO this could be a generic function that works for any format
+/// Read an `Rgba32FImage`.
+/// Assumes the reader is buffered. In most cases,
+/// you should wrap your reader in a `BufReader` for best performance.
+// TODO progress? load rect?
+pub fn read_as_rgba_image(read: impl BufRead + Seek) -> ImageResult<Rgba32FImage> {
+    let decoder = ExrDecoder::read(read, Some(true))?;
+    debug_assert_eq!(decoder.color_type(), ColorType::Rgba32F);
+
+    let (width, height) = decoder.dimensions();
+    let buffer: Vec<f32> = decoder_to_vec(decoder)?; // passing `true` to decoder results in 4 floats per pixel
+
+    ImageBuffer::from_raw(width, height, buffer)
+
+        // this should be the only reason for the "from raw" call to fail,
+        // even though such a large allocation would probably cause an error much earlier
+        .ok_or_else(|| ImageError::Limits(LimitError::from_kind(LimitErrorKind::InsufficientMemory)))
+}
+
+// TODO this could be a generic function that works for any format
+/// Read the file from the specified path into an `Rgb32FImage`.
+// TODO progress? load rect?
+pub fn read_as_rgb_image_from_file(path: impl AsRef<Path>) -> ImageResult<Rgb32FImage> {
+    read_as_rgb_image(BufReader::new(std::fs::File::open(path)?))
+}
+
+// TODO this could be a generic function that works for any format
+/// Read an `Rgb32FImage`.
+/// Assumes the reader is buffered. In most cases,
+/// you should wrap your reader in a `BufReader` for best performance.
+// TODO progress? load rect?
+pub fn read_as_rgb_image(read: impl BufRead + Seek) -> ImageResult<Rgb32FImage> {
+    let decoder = ExrDecoder::read(read, Some(false))?;
+    debug_assert_eq!(decoder.color_type(), ColorType::Rgb32F);
+
+    let (width, height) = decoder.dimensions();
+    let buffer: Vec<f32> = decoder_to_vec(decoder)?; // passing `false` to decoder results in 3 floats per pixel
+
+    ImageBuffer::from_raw(width, height, buffer)
+
+        // this should be the only reason for the "from raw" call to fail,
+        // even though such a large allocation would probably cause an error much earlier
+        .ok_or_else(|| ImageError::Limits(LimitError::from_kind(LimitErrorKind::InsufficientMemory)))
+}
+
+
+// TODO this could be a generic function that works for any format
+/// Write an `Rgb32FImage` to the specified file path, replacing any existing file.
+// TODO progress? load rect?
+pub fn write_rgb_image_to_file(path: impl AsRef<Path>, image: &Rgb32FImage) -> ImageResult<()> {
+    write_rgb_image(BufWriter::new(std::fs::File::create(path)?), image)
+}
```

Comment:
Let's skip adding this one line function for now. Shouldn't be too hard for people to write it themselves.